### PR TITLE
Enable unused variable check

### PR DESCRIPTION
**Pull request by bzsolt**
IoT.js-DCO-1.0-Signed-off-by: Zsolt Borbély zsborbely.u-szeged@partner.samsung.com

### CODE REVIEW

File: src/iotjs_binding.c [hunk 1]
```diff
@@ -167,7 +167,7 @@ void iotjs_jval_destroy(iotjs_jval_t* jval) {
 
 
 static void iotjs_jval_destroy_norelease(iotjs_jval_t* jval) {
-  IOTJS_VALIDATED_STRUCT_DESTRUCTOR(iotjs_jval_t, jval);
+  IOTJS_VALIDATED_STRUCT_DESTRUCTOR_VALIDATE(iotjs_jval_t, jval);
```
Comment (yichoi) [inline src/iotjs_binding.c]: Roughly, naming style in this patch is somewhat awkward.

Comment (bzsolt) [inline src/iotjs_binding.c]: Just suffixed the current ones, but I'm open for suggestions.

Comment (yichoi) [inline src/iotjs_binding.c]: I could not pick the best prefix term since I am not native English speaker :)

Comment (glistening) [inline src/iotjs_binding.c]: Suggested name ``IOTJS_VALIDATED_STRUCT_DESTRUCTOR_VALIDATE gives more information. But it is too long and some word are duplicated.

Basically, I don't like current prefix  ``IOTJS_VALIDATED_STRUCT``:
1) Too long. 
2) Also this structure does not validated, but validatable. But it seems validatable is not popular word. Sanity checkable would be better. But it is too long.

I cannot suggest better candidate that satisfy aboves.
What about landing unused variable warning only? And rename the name in another PR?

Comment (yichoi) [inline src/iotjs_binding.c]: Let's go with ``IOTJS_VALIDATABLE_STRUCT_DESTRUCTOR_VALIDATE``.
If we find better idea, then rename it in another PR.

Comment (bzsolt) [inline src/iotjs_binding.c]: Should I apply this form to the `IOTJS_VALIDATED_STRUCT_METHOD` too?

Comment (yichoi) [inline src/iotjs_binding.c]: yes, please

**Comment by bzsolt** [comment]
I've updated the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**